### ISSUE TITLE
[Bug Report][3.1.2] Autocomplete feels laggy

### ISSUE THREAD

**Issue by ThomasKientz** [vuetifyjs/vuetify#16318]
### Environment
**Vuetify Version:** 3.0.5
**Last working version:** 2.6.13
**Vue Version:** 3.2.45
**Browsers:** Chrome 108.0.0.0
**OS:** Mac OS 10.15.7

### Steps to reproduce
Use a VAutocomplete.

### Expected Behavior
When user focus, list menu to appear instantly, as with Vuetify 2.

### Actual Behavior
When user focus, list menu appear with a delay (100-200ms ?).

### Reproduction Link
[https://codepen.io/thomaskientz/pen/vYaOgwO?editors=1010](https://codepen.io/thomaskientz/pen/vYaOgwO?editors=1010)

### Other comments
Vuetify 2 exemple : https://codepen.io/thomaskientz/pen/dyjoNEb?editors=101

I think its a bug of the animation. With a VSelect it's fine, the animation is too harsh though imo.

Also, looks like its fine when VAutocomplete is instantiated with a value.

**Comment by KaelWD** [comment]
I don't see a delay

**Comment by mgd216** [comment]
I'm having similar issues with VAutocomplete, I have an items list of over 800 items with title and value props.  Typing in the autocomplete to filter has a sever delay.  However when I try and repdroduce the issue in Codepen, the Autocomplete filters fine:

https://codepen.io/mgd216/pen/poZyeBa?editors=1010

I'm using Vue 3.2.45, Vuetify 3.0.6 and confirued in Nuxt 3.0.0.  I'm trying to determine if Nuxt is layering some action over Autocomplete that is bogging it down.

**Comment by luiztsmelo** [comment]
I'm having a huge delay with v-autocomplete too with big lists, after migration. And I'm not using Nuxt.

**Comment by luiztsmelo** [comment]
@KaelWD change ` [...Array(10).keys()] ` to ` [...Array(5000).keys()] `, and you will see the huge delay.

**Comment by KaelWD** [comment]
I'm aware of problems with large numbers of items, trying to get #15585 in 3.1 to resolve that. @ThomasKientz didn't specify more than 10 items so idk if this is something else.

**Comment by ThomasKientz** [comment]
I was having issue with 10 elements only. I have pinned the exact version of vuetify in the repro, but you are right no delay / issue with animation anymore. Menu was hiding the label as well, not the case anymore. Maybe a cache issue on my side.Do you guys want I keep the issue open for large list ?Le 3 janv. 2023 à 15:37, Kael ***@***.***> a écrit :﻿
I'm aware of problems with large numbers of items, trying to get #15585 in 3.1 to resolve that. @ThomasKientz didn't specify more than 10 items so idk if this is something else.

**Comment by jaysonpotter** [comment]
I had submitted this same issue at https://github.com/vuetifyjs/vuetify/issues/16220 and repro steps were added and rejected because @KaelWD stated that running in dev was not an acceptable way to test performance.

**Comment by jaysonpotter** [comment]
Tested this with the 3.1.0 release and same results. 

https://codepen.io/jaysonpotter/pen/gOjLQRm

**Comment by drhouse82** [comment]
I agree that there is a big issue with large autocomplete opening times.
@KaelWD Merging #15585 did not really improve the situation, right?
For me, the codepen autocompletes of @mgd216 take 2-3s to open, in the last codepen of @jaysonpotter the autocompletes don't open at all.

**Comment by boukadam** [comment]
> I agree that there is a big issue with large autocomplete opening times. @KaelWD Merging #15585 did not really improve the situation, right? For me, the codepen autocompletes of @mgd216 take 2-3s to open, in the last codepen of @jaysonpotter the autocompletes don't open at all.

I'm using the last release v3.1.1 and I am having the same issue with VAutocomplete having data of around 40k items

**Comment by ThomasKientz** [comment]
@KaelWD still happening : you can reproduce with : https://codepen.io/thomaskientz/pen/PoBOvNN and compare with https://codepen.io/thomaskientz/pen/xxJPNGQ 

V3 feels laggy.

**Comment by ThomasKientz** [comment]
Even worse with `multiple` and 300 object items : 
v3 https://codepen.io/thomaskientz/pen/wvxPLex 
vs V2 : https://codepen.io/thomaskientz/pen/XWBzLeb

**Comment by ThomasKientz** [comment]
https://user-images.githubusercontent.com/10157845/213571697-cd9e9cea-8d04-4548-88f8-922749b343b5.mov

**Comment by KaelWD** [comment]
This is like five different problems all hitting at the same time:
 - connectedLocationStrategy calls itself twice (#16350)
 - useLocationStrategies also initialises itself twice, so updatePosition is called four times
 - all list items are always rendered (#2660)
 - list items render twice initially (#15215)
 - the way we retrieve prop defaults is inefficient and results in lots of proxy calls on patches

**Comment by yuwu9145** [comment]
> Even worse with `multiple` and 300 object items : v3 https://codepen.io/thomaskientz/pen/wvxPLex vs V2 : https://codepen.io/thomaskientz/pen/XWBzLeb

After playing around, although `locationStrategies` needs to be refactored, the major delay seems to be caused by "rendering" itself. In V3, it renders whole list once (which could cause huge delay for `nextTick` depends on how many items to render), but in V2, its initial rendering only renders first 20ish items and gradually renders more while scrolling.

**Comment by AniaKru95** [comment]
Is it possible to specify height of opening items?
Firstly, only few items will be loaded and after scroll rest of them?

**Comment by johnleider** [comment]
> Is it possible to specify height of opening items? Firstly, only few items will be loaded and after scroll rest of them?

We plan on integrating `v-virtual-scroll` once it's out of labs. We have identified the root of the issue and are working on a fix now.

**Comment by drich1402** [comment]
There is still a delay of more than one second if there are 1000 elements in the list. With vue2 there is a quick response. (vuetify 3.1.5)

**Comment by johnleider** [comment]
Considering it hasn't been fixed that sounds about right.

**Comment by simonklimek** [comment]
I also noticed that navigation drawer with Expand on hover animation is lagging, side menu has around 150 items in sub groups list items

**Comment by Ap3rtur3** [comment]
@johnleider Has there been progress on the issue?

**Comment by HendrikJan** [comment]
> @johnleider Has there been progress on the issue?

As I understood (correct me if I'm wrong), the solution will depend on virtual-scroll. Virtual scroll is in Vuetify Labs, so I would expect a fully functional solution only after virtual-scroll is out of labs.

**Comment by johnleider** [comment]
> As I understood (correct me if I'm wrong), the solution will depend on virtual-scroll. Virtual scroll is in Vuetify Labs, so I would expect a fully functional solution only after virtual-scroll is out of labs.

This is correct. Although 3.1.9 did release with quite a few micro optimizations that do help.

**Comment by johnleider** [comment]
> How would we implement Vuetify Labs virtual-scroll with v-autocomplete currently?

You can't unfortunately. We are getting virtual-scroll into 3.2 and will update the select components proper soon.

**Comment by johnleider** [comment]
The current plan is to release `v-virtual-scroll` in v3.2. Once we have some testing in it and improve keyboard support, we want to release an update for all select components to implement the virtual scroller.

**Comment by grasulkhuja** [comment]
> The current plan is to release `v-virtual-scroll` in v3.2. Once we have some testing in it and improve keyboard support, we want to release an update for all select components to implement the virtual scroller.

Will using `v-virtual-scroll` in some components like autocomplete, combobox, etc. be in v3.2 or v3.3?

**Comment by johnleider** [comment]
> Given that the 3.2 is now out, and that I didn't see any mention of the v-virtual-scroller being used in the components in the changelog, nor in the code, and given that this issue has been assigned to 3.3, I'd say we're out of luck, and still need to wait a bit, am I right ?
> 
> This is the main reason we were waiting for the 3.2, but unfortunately, we'll have to be patient again... 😞

The plan was always to release v3.2 with a fully released v-virtual-scroll then fix it as part of a bug patch.

**Comment by djokicpn** [comment]
I'm also still having issues with v-autocomplete in the 3.2.5 version. Even on 30 options, I have really annoying lag.

**Comment by johnleider** [comment]
> I'm also still having issues with v-autocomplete in the 3.2.5 version. Even on 30 options, I have really annoying lag.

If you're going to make a post saying "me too" please provide some sort of reproduction that differs from the current available. I highly doubt you're having lag issues with 30 items.

**Comment by djokicpn** [comment]
> > I'm also still having issues with v-autocomplete in the 3.2.5 version. Even on 30 options, I have really annoying lag.
> 
> If you're going to make a post saying "me too" please provide some sort of reproduction that differs from the current available. I highly doubt you're having lag issues with 30 items.

You are right. I just checked that I merged migration pull request few days ago. There are exactly 194 records and it feels laggy. I just tested, with 30 records it works fine, no lag at all

**Comment by jaysonpotter** [comment]
> > I'm also still having issues with v-autocomplete in the 3.2.5 version. Even on 30 options, I have really annoying lag.
> 
> If you're going to make a post saying "me too" please provide some sort of reproduction that differs from the current available. I highly doubt you're having lag issues with 30 items.

I check for releases every week and keep this test case of 5000 items current (Vuetify 3.3.1 as of today):
https://codepen.io/jaysonpotter/pen/gOjLQRm

**Comment by djokicpn** [comment]
> > > I'm also still having issues with v-autocomplete in the 3.2.5 version. Even on 30 options, I have really annoying lag.
> > 
> > 
> > If you're going to make a post saying "me too" please provide some sort of reproduction that differs from the current available. I highly doubt you're having lag issues with 30 items.
> 
> I check for releases every week and keep this test case of 5000 items current (Vuetify 3.3.1 as of today): https://codepen.io/jaysonpotter/pen/gOjLQRm

With 3.3.1 I can confirm that its working a LOT FASTER! loading 300 or 500 records is instant. on 1500 records its barely lagging.

**Comment by KaelWD** [comment]
Everyone here please try the nightly build from https://github.com/vuetifyjs/vuetify/pull/17265#issuecomment-1564103553

https://vuetifyjs.com/en/getting-started/installation/#nightly-builds

**Comment by drich1402** [comment]
The nightly build is awesome and perfecly responsive. Unfortunately, the 3.3.2 version is still slow and laggy. I will continue to be patient.

**Comment by hassanpdn** [comment]
The cause of the lag maybe for the large set of data, you can use v-intersect to implement infint scroll in v-autocomplete too. Why waiting for this issue here?

**Comment by KaelWD** [comment]
This isn't going to be merged until I get some positive feedback that it doesn't break anything else. There's 18 people in this thread and I've only heard from one.

**Comment by marr** [comment]
I don't see a performance difference between 3.3.2 and `@vuetify/nightly@3.3.0-master.2023-05-21` -- am I on the right nightly version?

**Comment by KaelWD** [comment]
No: https://github.com/vuetifyjs/vuetify/pull/17265#issuecomment-1567258080

**Comment by efthymiou** [comment]
Auto complete in 3.3.2-pr-17265.4b7bc69 is blazing fast and responsive with my 10k entries test case. My other test examples with a advanced (multiple overriden slots, html option, icons)  usage, seem to work fine.

LGTM

**Comment by drennenb** [comment]
Went searching for performance issues with V3 autocomplete and found this thread.  My autocomplete has 820 items in it using custom templates.  In V2 its instant to open/display.  In 3.3.2 and the nightly build listed above it takes around 2 seconds to open.  There is no difference between 3.3.2 and the nightly that I can discern.

**Comment by KaelWD** [comment]
Make sure you have it installed correctly, it should replace the normal vuetify version.

**Comment by drennenb** [comment]
I don't think I was using vuetify/nightly@3.3.2-pr-17265.4b7bc69.  Thanks for the push to try it again. This performs much better, instantaneous.  Thanks again.

**Comment by SkyaTura** [comment]
@KaelWD when the items are loaded while the menu is open it won't observe for changes.

I have a component that loads the items only after the input is focused for the first time. The scroll is getting the correct height, but only the first few items are rendered.

If I close the menu and open again (with the items already loaded) it works just fine.

**Comment by KaelWD** [comment]
Should be fixed now, try the new build

> is it possible to create a "list" slot

#15014

**Comment by jaysonpotter** [comment]
Thank y'all for the updates, but still appears to lag: https://codepen.io/jaysonpotter/pen/gOjLQRm

**Comment by SkyaTura** [comment]
@KaelWD it's flying now, nicely done!

---

@jaysonpotter your pen is at the wrong version. You should test with `@vuetify/nightly@3.3.3-pr-17265.321510f` since this PR is not merged yet.

Here is your test with the latest build on that branch https://codepen.io/SkyaTura/pen/zYMxQjP

**Comment by jaysonpotter** [comment]
> @KaelWD it's flying now, nicely done!
> 
> @jaysonpotter your pen is at the wrong version. You should test with `@vuetify/nightly@3.3.3-pr-17265.321510f` since this PR is not merged yet.
> 
> Here is your test with the latest build on that branch https://codepen.io/SkyaTura/pen/zYMxQjP

Thank you @KaelWD ! I concur that the nightly@3.3.3-pr-17265.321510f is working fantastically and as expected. 🎉

**Comment by gabaum10** [comment]
I'm not sure if I configured it wrong, but I'm still seeing some pretty significant performance issues with our autocompletes using the nightly here.  Is there some way I can confirm the changes?  It looks like the nightly is already compiled when I pull it in so it's hard to tell.

I can do a side by side in our staging environment with what's currently live vs our Vue/Vuetify 3 upgrade branch it's still very much an issue.  I can't seem to replicate it in a sandbox though, so it's really hard to tell.  We're using the `return-object` prop, which I'm not sure whether that's the source of the issues or not.

I guess push comes to shove I can test once it's officially merged in to take the PR nightly guesswork out of the equation haha.

**Comment by gabaum10** [comment]
This is what I'm seeing with the vuetify 2.x implementation
<img width="964" alt="image" src="https://github.com/vuetifyjs/vuetify/assets/1561245/06fecbcb-9fcd-4a4d-b3fb-704147ae24dc">


And this is what I see in the Vuetify 3 nightly for this PR:
<img width="1020" alt="image" src="https://github.com/vuetifyjs/vuetify/assets/1561245/215ed57a-7842-4e02-bf42-ebdcf61e936c">

Notice the massive scrollbar, which means all 3000 of the items are being populated into the menu.

**Comment by gabaum10** [comment]
Actually even @jaysonpotter 's pen above is demonstrating the same behavior and not correctly paging.  We just have a more complicated data set so it's getting bogged down more.

**Comment by SkyaTura** [comment]
@gabaum10 can you provide a reproduction link ?

**Comment by gabaum10** [comment]
After looking closer, I have to be using the wrong instance of Vuetify.  In the Pen above, it's correctly using the virtual scroller, but on my example in my code it's rendering every single list item in the menu.  I'm not sure if that's because of our `v-autocomplete` configuration or it's not using the nightly code.

**Comment by acnowland** [comment]
@SkyaTura , I am on this same project and I made sure that we are running the nightly build on our local dev server, we are indeed on master.2023-06-09. In the playground attached you will find an example in a playground. There are 3 autocomplets with large object data sets, one is set to have menu open so it can be inspected. If you inspect the page and look at the dom you will notice that all 2000 items are being rendered to the dom. This is what is happening when we are opening out v-autocompletes as well with the menu prop off. This can be observed when just opening and closing any of the autoselects there. For better or worse we operate with large data sets which exceed 2000 objects in an array so this is our  issue. 

https://play.vuetifyjs.com/#eNrtVt1u2jAUfhUrN1CVJAymScto1Um93ROUqjLkBNw5dmQ7bLTi3ffZCTQEJqHeTJoKgtjn9zufjw88vEbfqyrZ1BRl0cxRWUnu6HauGJtt4iU3eVg3O60cF4oMK2Qt9oqgMvrXYdvaSoaPvZlHnybzqKMMal47vdRIRo66KsY2calzkvDbcFmTnUfH+kwApA9LkkpS7sRgKYkbvpC9uJIvQtR7KngtXd/LR42dcJJgo3hJZw0CJBiIvK/OSVnhttChrEIbFxD0oa1FZY9FKKHuSYBOgJh+Vdr6kPFJjNtZeszn+7nGQf3fZP87aqcf1F5MrVdhdrzNF+zfBkyjbSdRM6i8pJlUs7QzwrC1SyMqF6zod4VyUGDgjb020XLueMaGV+zmlg1bWUOSzdjDoNB6MGKDBTf+UYiXl7Ct8Xwc7a2bE86YqqXsCSc96f6wEbvvPz0y3V21iwXhFOiHrpUDygNCCNkQvDHFbtj4Gx4zNhmP/er6umPHWJqypSHuhFoxvXimpWNrMp2DApnWeRUidfzAQp6xAeRPIo8H7JqpA2L/8s0FvX88nWoLXgq5hb5ZnLHgK++O73O6PDdkwdKgXZ3Y7N6Wbi1ssic2qWq7HgL01d6gtdwFZ2xm6aEpolEkhfppk2erFX7/QvXofIvLiBNqHOfR2rnKZmm6zBUscavFxiSKXKqqMr3DT6cTxTZVYrV2cns3TfCOS24dmWQynkzj8Zd4/DXNhXVpaxzjFtsk5PFJHudqByyi9B0K16oHqFEEUO0BYazUuJbZJehSg94RJcW5LgO6zw2WjjghW8YL3DELzM9I1DId0njAF6Z6JxHIfpI1BZQNmdiQytGu5tJie27dgnuqk6LbBvFHcS+41Kuzf4t6Q6Y7YjrNFe1GUUiNYH8hInr8AwIIyBY=

**Comment by SkyaTura** [comment]
@gabaum10 have you tried installing with `npm install --save vuetify@npm:@vuetify/nightly@3.3.3-pr-17265.321510f`?

---

I did some experiments on the links sent by @gabaum10 @acnowland and it is, indeed, opening and scrolling as expected. However, when pushing things further to explore the limits on this matter, I found that after typing something and clearing back, it freezes for a couple instants before actually clear the input.

As far as I went, the PR seems fine, it is definitely way better than current release, and I don't really think that autocompletes should be used on 10k arrays, however, this may be a point of attention.

**Comment by gabaum10** [comment]
Yeah as @acnowland mentioned, it's working much better on our love example.  This should be good.  We'll run the nightly for now as we finish the migration and let you know if we see anything weird.

Do you guys know when this fix will get pushed out?

**Comment by KaelWD** [comment]
If anyone else is having problems with installation, make sure to
1. *Replace* vuetify with the nightly build, do not install them both. This should be done with a package alias so all the import paths are unchanged, see the diff under https://vuetifyjs.com/en/getting-started/installation/#nightly-builds for an example.
2. Use the correct build, it should contain `<version>-pr-17265.<hash>` not `<version>-master.<date>`
---

> after typing something and clearing back, it freezes for a couple instants

@SkyaTura Do you have some code and exact steps to take to reproduce this?

---
>  when this fix will get pushed out?

@gabaum10 When I'm satisfied that it isn't significantly broken.

**Comment by SkyaTura** [comment]
> @SkyaTura Do you have some code and exact steps to take to reproduce this?

I forgot to send the reproduction haha. Basically it consists on a simple autocomplete with 20k+ items. Then you open the menu, type something and erase a couple times. Each time you manually clear the input it takes longer on the last character.

https://play.vuetifyjs.com/#eNrtVt1u2jAUfhUrN1CVJAXWTcqg6qS+RakqEzvgzrEj22GjFe++z06AELhAlXbXILB9vuPz8x37hOeP6FdVJZuaR1k0c7ysJHX8YaEImW3inBoW5s1KK0eF4oYUshZ7IEBG/zksW11J8LXzRTSeLKIOGGBaO51rOOOOdyFCNnGpGZfYt6Gy5nYRneKZQJDeLJe85MqdKeSSU0OXsmdX0mWw+sQLWkvX3+Wtxk44yaGjaMkvKoSQoCBYH2ZcWeG2wJBWoY0LEfRDW4vKnopKxCJAQz8Hbb2B+GzHwyw9Ze/zzKIsX9T+H2qnX9ReTa2H0CmO3QTrYztp0LbvNG3JS5q+NEs7DQtLmxtRuaDF/1ZIBwkG3shHY41RRzMyvCHzBzJsZQ1JNiPPg0LrwYgMltT4oRDv72FZY3wZ7bWbCmdE1VL2hJOedF9s2O7vn56o7m7aSalr5ThDhIfokL11TYxkDkN7OcpFhuCSKIjHPzHM5uT+Dg/mt7cdC4SkKckNp06oFdHLN547suamU77GCSDY6uwDNywjA8hfBYsH5JaoQx7+8UcOuB9ez9GClkJugTeTCxp05bfj9xLGmOEW3A3a2ZnO7jgN9CRVbddDBHuzBw4abi1ssi8Hcgz6LffBIjRn6eH8RKNICvXbJm9WK7wYAyW4JBb3FsVsNi6itXOVzdI0ZwqaaABiYxLFXaqqMn3EO9WJYpsqsVo7uX2cJvjElYnHPybf75PpZHw/vitSJqxLW90Y990mwY33gVrvEIoo/VmOS1r14mmAEFNbNDSgGhc4uya41OCsiZLHTJchuG9NLB1xwm0ZL3EbLTcwsoha9oMbH/CVrj7HA5yfOU0RyYab2HDFcILNtbn2tnXz7UFnObfHw1fiSVCpVxf/LfW6UbcXdY5WtBtFwTWMXeYhevkHj1HN7w==


https://github.com/vuetifyjs/vuetify/assets/6800723/53298b6d-a77c-4ae0-b01c-664596eff406

**Comment by johnleider** [comment]
At 20k it seems fine: https://play.vuetifyjs.com/#eNrtVt1u2jAUfhUrN1CVJAXWTcqg6qS+Rakqk5yAO8eObIeNVrz7vjgBQuAC9WbStCCwfb7j8/Md+4Tnj+BHWUabioIkmDkqSskdPSwUY7NNmHKT+Xmz0spxociwXFZiD3jI6F+HZasrGb52vgjGk0XQAT3MK6dTDWfkqAsxtgkLnZHEvg2XFdlFcIonAkHWZklSQcqdKaSSuOFL2bMr+dJbfaKcV9L1d9VWQyecJOgoXtBFBR8SFETWhzNSVrgtMKSVa+N8BP3Q1qK0pyLiKzKnogLhCTDTT0vb2mZ4ZuRhFp8S+nmyUal/mu2/SO30P7VXU1tDaB7HBoP1scM0aNuKmk5VS5pWNYs7PQxLmxpROq9Fv0ukgwQ9b+yjsZZxxxM2vGHzBzZsZQ1JNmHPg1zrwYgNltzUQy7e3/2ywvgy2ms3FU6YqqTsCSc96b7YsN3fPz1R3d20k0JXylGGCA/RIXvrmhjZHIb2cpSLDcElUxCPv2OYzdnkDg/mt7cdC4zFMUsNcSfUiunlG6WOrcl0ytc4AQRbnX3gJkvYAPJXkYUDdsvUIY/6qY8c8Hp4PUdzXgi5Bd5MLmigHwLG7yUsywxZcDdoZ2c6u+PU0xOVlV0PEezNHjhouLWw0b4cyNHrt9x7i9CcxYfzE4wCKdRPG71ZrfCu9JTgkljcWxSz2bgI1s6VNonjNFPQRAMQGxMpcrEqi/gRr1kn8m2sxGrt5PZxGuETliYcf5t8vY+mk/H9+C6PM2Fd3OqGuO828m5qH6j1DqGIoj7LYcHLXjwN4GNqi4YGVOECJ9cEFxucNVFQmOnCB/eliaUjjsgW4RK30ZKBkUXQsu/d1AFf6epzPMD5mdMYkWzIhIZUhhNsrs21t62bbw86y7k9HnUlngSXenXxD1SvG3V7UedoBbtR4F3D2GUegpc/boLSOg==

I feel like 50k is an extreme example.

**Comment by gabaum10** [comment]
I feel like even 20k is an extreme example.  In my testing, this seems like a great fix. 👍

**Comment by KaelWD** [comment]
Filtering should be more than 30x faster now for 50k items, 240ms -> 6.5ms. Even 500k items takes under 200ms now.

**Comment by acnowland** [comment]
@KaelWD  so it is working better but not great in our application.. now I cant replicate it though but we are using the PR build  b/c it load smuch quicker, but the issue is on scroll. The items load quick initially but as we scroll there is a delay from when you scroll and when they are rendered. I was doing some performance testing and unsure if related, but if you compare the old in v2 vs v3 we can see that in the PR version it is literally firing off THOUSANDS of event listeners as you scroll. While in the old it fires only 2 each time you scroll. I imagine this can lead to some performance issues. Not sure if fixed already or not. 

v2:
<img width="849" alt="Screenshot 2023-06-12 at 9 38 55 AM" src="https://github.com/vuetifyjs/vuetify/assets/75752080/0be39604-c4a8-4910-bdd3-b6a9b87e0019">

v3: 
<img width="858" alt="Screenshot 2023-06-12 at 9 36 04 AM" src="https://github.com/vuetifyjs/vuetify/assets/75752080/bedf82c6-d59a-4538-a55e-b557444d02d2">


v2 was taken from our prod application so there is other stuff going on there, v3 was recorded though in the playground on the pr17265 branch

**Comment by acnowland** [comment]
any updated on my concern? its an awful lot of listeners being triggered on scroll with the new implementation vs the old. We are also seeing some kind of weird interaction with building our project with that nightly PR included with this fix. For w/e reason on a local dev server we see the full list of items being rendered as you scroll, but after we deploy it only renders 30-50% of the items. I can inspect the autocomplete after deploy and see the :items have been passed in, and you can even search for the hidden items and they appear, but as you scroll they do not render.. I cannot for the life of me replicate on a playground though and also not sure if its build related on our side or something from vuetify side tbf. 

```
vuetify: "npm:@vuetify/nightly@3.3.3-pr-17265.321510f"
```
Thats the build we are using which I think is correct. I realize w/o a replication its tough, but figured it should be brought to attention

**Comment by KaelWD** [comment]
I'm not seeing thousands. Each step here is one scrollwheel notch, then I closed the manu, waited a couple seconds, and triggered GC:
![Screenshot_20230615_155801](https://github.com/vuetifyjs/vuetify/assets/16421948/84fbd036-94ce-44b2-9cfe-cdd09d087c74)


This is scrolling all the way to the end, back to the top, then triggering GC:
![Screenshot_20230615_141851](https://github.com/vuetifyjs/vuetify/assets/16421948/ddf02ce1-b4b4-471b-acf0-af453eedf38e)

The reason it doesn't drop as soon as the menu closes is probably just v8 being efficient and not cleaning up until it actually runs out of memory.

The retained memory seems to be effect dependency arrays in the scope of this vue internal function I think?
![Screenshot_20230615_155444](https://github.com/vuetifyjs/vuetify/assets/16421948/040c4f25-b715-414f-b1ca-d857bf6551ea)
![Screenshot_20230615_155213](https://github.com/vuetifyjs/vuetify/assets/16421948/9128df22-c705-44a1-baec-118a51b153f9)

> after we deploy it only renders 30-50% of the items

I actually reproduced this by accident trying to see if the memory behaviour was different in a production build (because that closure wouldn't exist), looks like it's getting window as the scrollable parent instead of the list for some reason.

**Comment by KaelWD** [comment]
`vm.ctx.$el` is only defined in dev: https://github.com/vuejs/core/blob/020851e57d9a9f727c6ea07e9c1575430af02b73/packages/runtime-core/src/component.ts#L572-L576

This is the second test but in a prod build, seems to collect partway through scrolling and doesn't retain anything after the manual GC:
![Screenshot_20230615_162041](https://github.com/vuetifyjs/vuetify/assets/16421948/31c498b0-5544-4138-9a53-7edde84970ce)
It does peak at 50k listeners but there's 50k items so that makes sense.

**Comment by KaelWD** [comment]
It might be possible to shift the window based on scroll velocity though.

**Comment by johnleider** [comment]
> It might be possible to shift the window based on scroll velocity though.

Such as increasing the buffer on either side depending upon how fast the user is scrolling?

**Comment by gabaum10** [comment]
Hey is there any way we can get a timeline update for this?  It's one of the last major pieces we're waiting for from the v2 -> v3 migration.  We also would like to make use of the two patches the PR nightly is behind for this as the fixes within address some of the other issues we've had to work around.

**Comment by johnleider** [comment]
> Hey is there any way we can get a timeline update for this? It's one of the last major pieces we're waiting for from the v2 -> v3 migration. We also would like to make use of the two patches the PR nightly is behind for this as the fixes within address some of the other issues we've had to work around.

This is the timeline: https://github.com/vuetifyjs/vuetify/pull/17265

**Comment by gabaum10** [comment]
@johnleider is there any way we can get an up to date nightly for that PR published?  The one we've been testing is weeks old.  

There is absolutely no way we can wait around until September for 3.4 to roll out the upgrade.  As is, all additional development is currently on hold while we iron out the remaining Vuetify 3 issues.

**Comment by johnleider** [comment]
It's marked for v3.4 but it's based off of master. Once @KaelWD collects enough feedback to feel comfortable with shipping it, we'll merge and release as part of a regular weekly patch.

**Comment by gabaum10** [comment]
Great, thanks!  As soon as we can get an updated nightly build off that PR to test the other reported issue it'll probably be good on our side.  But we're waiting to confirm.

**Comment by KaelWD** [comment]
@gabaum10 I've published another one with the latest changes

**Comment by Rusinas** [comment]
@KaelWD thank you for you work, now it's much better. Unfortunately, virtual scroll doesn't work in production for some reason. After `vite build` it looks like this:
![image](https://github.com/vuetifyjs/vuetify/assets/42975143/615bd910-a021-450d-855e-095d6108b0a4)

**Comment by gabaum10** [comment]
> @KaelWD thank you for you work, now it's much better. Unfortunately, virtual scroll doesn't work in production for some reason. After `vite build` it looks like this: ![image](https://user-images.githubusercontent.com/42975143/248251898-615bd910-a021-450d-855e-095d6108b0a4.png)

This is the same issue we were hoping that was fixed by the changes in the PR.  I guess it's not though.  I'm not entirely sure how to replicate it, but we will work on trying to come up with an example in the playground.

**Comment by gabaum10** [comment]
@KaelWD FYI, I'm not seeing the issue reported above.  I suspect they are still using the old version of the build.  As far as I can tell, all our issues are now resolved.  This is great.  Thanks for your work on this stuff!

**Comment by acnowland** [comment]
Yea this looks great now! Thank you all! 🙏🏻

**Comment by Rusinas** [comment]
@gabaum10 the build I'm using is `nightly@3.3.3-pr-17265.321510f`. Did I miss an update?

**Comment by gabaum10** [comment]
@Rusinas yes.  `"vuetify": "npm:@vuetify/nightly@3.3.5-pr-17265.042d482"`
